packages/protocol/src/abstracts/BaseRouter.sol

```diff
@@ -100,8 +100,35 @@ abstract contract BaseRouter is SystemAccessControl, IRouter {
   }
 
   /// @inheritdoc IRouter
-  function xBundle(Action[] calldata actions, bytes[] calldata args) external payable override {
-    _bundleInternal(actions, args, 0);
+  function xBundle(
+    Action[] calldata actions,
+    bytes[] calldata args
+  )
+    external
+    payable
+    override
+    nonReentrant
+  {
+    _bundleInternal(actions, args, 0, Snapshot(address(0), 0));
+  }
+
+  /// @inheritdoc IRouter
+  function xBundleFlashloan(
```

Comment:
I would like to clarify that this would apply only if the same `IFlasher` is being used. If the nested flashloan inside a previous flashloan calls a different `IFlasher` it could still be possible to reenter. right?

Reply:
Added  modified nonreentrant so that there can only be one flashloan call in a bundle regardless if different IFlasher attempts to call.